sound/soc/sof/ipc4-topology.c

```diff
@@ -556,6 +556,9 @@ static void sof_ipc4_widget_free_comp_dai(struct snd_sof_widget *swidget)
 	if (!dai)
 		return;
 
+	if (!dai->private)
+		return;
```

Comment:
sof_ipc4_widget_setup_xxx() is called in sequence. If sof_ipc4_widget_setup_comp_dai() is not called, it will be NULL. But even sof_ipc4_widget_setup_comp_dai() is not called, sof_ipc4_widget_free_comp_dai() will still be called. This is the same situation as other sof_ipc4_widget_free_xxx() callbacks.

Reply:
@ujfalusi Yes, I will free dai if `dai->private == NULL` . And I will refine the error handling in the `sof_ipc4_widget_setup_comp_dai()`